test/tx_loc_modes3.ml

```diff
@@ -0,0 +1,32 @@
+(* Location is in obstruction_free mode
+   Each Thread increments the value at a and reads a in a loop*)
+
+module Loc = Kcas.Loc
+module Mode = Kcas.Mode
+module Tx = Kcas.Tx
+
+let a = Loc.make ~mode:Mode.obstruction_free 4
+let read_loc loc = Tx.(return @@ Loc.get loc)
```

Comment:
I think I need to add warnings to the documentation on this, but unfortunately using operations from the `Loc` module is not how transactions should typically be created.  You should use the operations from the `Tx` module.

```diff
-let read_loc loc = Tx.(return @@ Loc.get loc)
+let read loc = Tx.get loc
```

`Loc.get loc` just immediately reads the value of `loc`.

`Tx.get loc`, on the other hand, returns a transaction that, when committed, reads the value of `loc` within the transaction.

There are situations when one might want to use `Loc.get` inside a transaction, but those are cases that need very careful analysis and understanding of how things work.